subsystem/src/main/java/org/wildfly/subsystem/resource/capability/CapabilityReferenceListAttributeDefinition.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright The WildFly Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.wildfly.subsystem.resource.capability;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+import org.jboss.as.controller.AttributeMarshaller;
+import org.jboss.as.controller.AttributeParser;
+import org.jboss.as.controller.ListAttributeDefinition;
+import org.jboss.as.controller.OperationContext;
+import org.jboss.as.controller.OperationFailedException;
+import org.jboss.as.controller.PrimitiveListAttributeDefinition;
+import org.jboss.as.controller.operations.validation.ModelTypeValidator;
+import org.jboss.as.controller.registry.AttributeAccess.Flag;
+import org.jboss.dmr.ModelNode;
+import org.jboss.dmr.ModelType;
+import org.wildfly.subsystem.resource.SimpleResource;
+import org.wildfly.subsystem.resource.ResourceModelResolver;
+import org.wildfly.subsystem.service.ServiceDependency;
+
+/**
+ * An list attribute definition referencing a capability.
+ * Resolves directly to a list of {@link ServiceDependency} via {@link #resolve(OperationContext, ModelNode)}.
+ */
```

Comment:
Maybe specify here that is only valid for a String List attribute definition? Otherwise the ModelType could be extracted as a generic value so any code using it can specify the model type, or maybe the intention was to make this extends `StringListAttributeDefinition`?

Reply:
The element ModelType is meant to be an implementation detail.
I would have extended StringListAttributeDefinition, but it (or rather its builder) is not extensible.

Reply:
This extends StringListAttributeDefinition now.